tools/const_finder/test/fixtures/lib/consts.dart

```diff
@@ -36,6 +39,30 @@ class IgnoreMe {
   final Target target;
 }
 
+class StaticConstInitializer {
+  static const List<Target> targets = <Target>[
```

Comment:
Consider adding a test with deeper nesting to test the caching optimization.

Reply:
@alexmarkov  - can you help suggest a test for this, or point me to one in the Kernel compiler?

Reply:
I'm not entirely sure what we'd test here either - we don't have benchmarks for this and I'm not sure we want to set them up, and otherwise I guess we could expose number of visits done, but that puts us at the mercy up upstream changes in the kernel pretty tightly...

Reply:
We have Dart tests for such cases:

https://github.com/dart-lang/sdk/blob/master/tests/language_2/canonicalize/hashing_memoize_array_test.dart
https://github.com/dart-lang/sdk/blob/master/tests/language_2/canonicalize/hashing_memoize_instance_test.dart

Traversal of such constants without caching will time out for any reasonable time limit.

Reply:
Perfect, thanks - I took one of those and basically copied it into here with slight modifications for Flutter style.